### PR TITLE
Feat/v2 for multi endpoint mapping

### PR DESCRIPTION
**Pull request by pauldambra**
## Changes

see https://github.com/PostHog/apps/issues/11

* adds a new method for sending events to the salesforce sink that uses a JSON mapping to allow multiple endpoints
* implements property filtering which is in config but doesn't actually happen
* adds some tests and structure

## Checklist

-   [x] Tests for new code

### CODE REVIEW

File: plugin.json [hunk 1]
```diff
@@ -64,6 +64,13 @@
             "hint": "turn on debug logging to get _much_ more logging",
             "choices": ["debug logging on", "debug logging off"],
             "default": "debug logging off"
+        },
+        {
+            "key": "eventEndpointMapping",
+            "name": "Event endpoint mapping",
+            "type": "string",
+            "hint": "⚠️ For advanced uses only ⚠️ Allows you to map events to different SalesForce endpoints",
```
Comment (lharries) [inline plugin.json]: Should we give an example? (Or add an example to this readme and link here)

File: src/eventSinkMapping.ts [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+}
+
+export const validateEventSinkConfig = (
+    eventMapping: EventToSinkMapping | null,
+    config: SalesforcePluginConfig
+): void => {
+    if (eventMapping !== null) {
+        Object.entries(eventMapping).map((entry) => {
+            const eventSink = entry[1]
+            debugger
```
Comment (lharries) [inline src/eventSinkMapping.ts]:
```suggestion
```

Comment (pauldambra) [reply src/eventSinkMapping.ts]: homer-hedge

File: README.md [hunk 3]
```diff
@@ -1,26 +1,56 @@
...
+# PostHog SalesForce App
 
-[![npm package](https://img.shields.io/npm/v/posthog-plugin-hello-world?style=flat-square)](https://www.npmjs.com/package/posthog-plugin-hello-world)
 [![License: MIT](https://img.shields.io/badge/License-MIT-red.svg?style=flat-square)](https://opensource.org/licenses/MIT)
 
-This is an exemplary PostHog plugin. It adds property `"greeting"` to every event, with a configurable value (by default: `"Hello world!"`).
+This is a PostHog export app. Use it to send properties from selected events to SalesForce. For example to create a lead when a customer books a demo.
 
-Use it as a base for your own plugins!
+See `plugin/json` for definition of accepted config
```
Comment (pauldambra) [inline README.md]:
```suggestion
```

**Comment by pauldambra** [comment]
closing in favour of the PostHog managed repo https://github.com/PostHog/salesforce-plugin/pull/8

**Pull request closed without merging**